draft-ietf-sidrops-aspa-verification.xml

```diff
@@ -541,92 +602,117 @@ hop(AS(i), AS(j)) =  / Else, "Provider+" if the VAP-SPAS entry
-	Implementation of those procedures in addition to ASPA-based AS_PATH verification is encouraged.
+       For any route with an Invalid AS_PATH, the cause of the Invalid state SHOULD be logged for monitoring and diagnostic purposes.
+       The cause of the Invalid state can be recorded in the form of listing the AS hops which were evaluated by the hop-check function to be &quot;Not Provider+&quot;.
+       The logging router, however, cannot necessarily determine the AS that caused the route leak.
+       For any route with an Unknown AS_PATH, the cause of the Unknown state MAY be logged for monitoring and diagnostic purposes.
+       If logging is done for the Unknown case, the cause can be recorded in the form of listing the AS hops which were evaluated by the hop-check function to be "No Attestation" or "Not Provider+".
       </t>
-    </section>
+   </section>
+  <!--
+  <section title="Verification and Mitigation at Egress eBGP Router" anchor="egress">
+    <t>
+      <strong>Verification:</strong> The procedures for AS_PATH verification (<xref target="verif"/>) are also applicable at egress eBGP routers for preventing an AS from sending routes with Invalid AS_PATH to its eBGP neighbors (see <xref target="RFC8893"/> for a comparable idea for the RPKI-ROV case).
+      An egress eBGP router MUST add the AS of the router's BGP configuration (see <xref target="RFC8481"/> <xref target="RFC8893"/>) to the received AS_PATH (received in iBGP) and then apply the path verification procedures.
+      When redistributing into BGP from any source (e.g., IGP, iBGP, or from static or connected routes), there is no AS_PATH in the input route.
+      In such cases, the egress eBGP router MUST use the AS of the router's BGP configuration to form the AS_PATH for verification (see <xref target="RFC8481"/>).
+   </t>
+   <t>
+     <strong>Mitigation:</strong> Again, mitigation recommendations are provided here with the understanding that the deployed mitigation policy is set by network operator discretion.
+     If the outcome of applying the upstream algorithm (<xref target="Upflow"/>) is Invalid, then the route SHOULD NOT be propagated to a transit provider, lateral peer, RS (local AS has RS-client role), or RS-client (local AS has RS role).
+     If the outcome of applying the downstream algorithm (<xref target="Downflow"/>) is Invalid, then the route SHOULD NOT be propagated to an eBGP neighbor regardless of its BGP role (<xref target="role"/>).
+   </t>
+  </section>
+  -->
+   <section title="Only to Customer (OTC) Attribute" anchor="otc">
+   <t>
+    While the ASPA-based AS_PATH verification method (<xref target="verif"/>, <xref target="impl-verif"/>) detects and mitigates route leaks that were
+    created by preceding ASes listed in the AS_PATH, it lacks the ability to prevent route leaks from occurring at the local AS.
+    The use of the Only to Customer (OTC) Attribute <xref target="RFC9234"/> fills in that gap.
```

Comment:
I understand the idea of telling people that `OTC` and `ASPA` are orthogonal and both should be used,
but I don't think that OTC prevents route leaks at the local AS (route leaks are always between ASs).

Reply:
>I don't think that OTC prevents route leaks at the local AS (route leaks are always between ASs).

Good catch. The wording is not right. We didn't mean a route leak internal to the AS. 
s/ it lacks the ability to prevent route leaks from occurring at the local AS / it lacks the ability to prevent the local AS from initiating a route leak towards its neighbor /
OTC is used as tag from the ingress to the egress and is used on egress to prevent the local AS from being a leaker. RFC 9234 wording is as follows: "In the case of prevention at the local AS, the presence of an OTC Attribute indicates to the egress router that the route was learned from a Peer, a Provider, or an RS, and it can be advertised only to the Customers."

Reply:
> s/ it lacks the ability to prevent route leaks from occurring at the local AS / it lacks the ability to prevent the local AS from initiating a route leak towards its neighbor /

I think this change helps. Even OTC does not detect all route leaks initiated by the local AS (even if the provider AS has RFC9234 implemented) since it depends on the transitive nature of the `OTC` attribute. But this is out of scope here.

The goal is to tell everyone to implement both :)

Reply:
@cjeker @mitradir 
>Even OTC does not detect all route leaks initiated by the local AS (even if the provider AS has RFC9234 implemented) since it depends on the transitive nature of the OTC attribute. 

Claudio: That is not correct. Assume the local AS is doing OTC. If the OTC attribute is not present on a route received from a provider or lateral peer, then the local sets it on eBGP ingress and then the route will not be propagated to a provider or lateral peer on eBGP egress.